packages/data/src/dispatch.ts

```diff
@@ -28,8 +23,10 @@ import defaultRegistry from './default-registry';
  * ```
  * @return Object containing the action creators.
  */
-export function dispatch< T extends StoreDescriptor< AnyConfig > >(
-	storeNameOrDescriptor: string | T
-): ActionCreatorsOf< ConfigOf< T > > {
```

Comment:
While testing in another repo, I came across an issue where `dispatch( 'store-name' )` had the return type `ActionCreatorsOf...`. This return type doesn't work, because we aren't using the object format of the store name. `useDispatch` handles this correctly (the return type should be unknown or similar, since we don't know any metadata about the store).